apps/resource_app.go

```diff
@@ -0,0 +1,181 @@
+
+	// Computed fields
+	for _, p := range []string{"active_deployment", "app_status", "compute_status", "create_time", "creator",
+		"default_source_code_path", "pending_deployment", "service_principal_id", "service_principal_name",
+		"update_time", "updater", "url"} {
+		s.SchemaPath(p).SetComputed()
+	}
+
+	// SuppressDiff
+	s.SchemaPath("source_code_path").SetCustomSuppressDiff(common.WorkspacePathPrefixDiffSuppress)
+
+	return s
+}
+
+var appSchema = common.StructToSchema(appStruct{}, nil)
+
+func ResourceApp() common.Resource {
+	return common.Resource{
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var createApp appCreateStruct
+			var createAppDeployment apps.CreateAppDeploymentRequest
+			common.DataToStructPointer(d, appSchema, &createApp)
+			common.DataToStructPointer(d, appSchema, &createAppDeployment)
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+
+			// create the app, which does not require the source code path yet
+			wait, err := w.Apps.Create(ctx, createApp.CreateAppRequest)
+			if err != nil {
+				return err
+			}
+			app, err := wait.GetWithTimeout(d.Timeout(schema.TimeoutCreate) - deleteCallTimeout)
+			if err != nil {
+				log.Printf("[ERROR] Error waiting for app to be created: %s", err.Error())
+				_, nestedErr := w.Apps.DeleteByName(ctx, createApp.Name)
+				if nestedErr != nil {
+					log.Printf("[ERROR] Error cleaning up app: %s", nestedErr.Error())
+				}
+				return err
+			}
+			// now deploy the app, using the source code path
+			createAppDeployment.AppName = app.Name
+			waitDeploy, err := w.Apps.Deploy(ctx, createAppDeployment)
+			if err != nil {
+				return err
+			}
+			_, err = waitDeploy.GetWithTimeout(d.Timeout(schema.TimeoutCreate) - deleteCallTimeout)
+			if err != nil {
+				log.Printf("[ERROR] Error waiting for app to be deployed: %s", err.Error())
+				_, nestedErr := w.Apps.DeleteByName(ctx, createApp.Name)
+				if nestedErr != nil {
+					log.Printf("[ERROR] Error cleaning up app: %s", nestedErr.Error())
+				}
+				return err
+			}
+			d.SetId(app.Name)
+			return nil
+		},
+		Read: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			app, err := w.Apps.GetByName(ctx, d.Id())
+			if err != nil {
+				return err
+			}
+			return common.StructToData(appStruct{App: *app}, appSchema, d)
+		},
+		Update: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			var update appUpdateStruct
+			common.DataToStructPointer(d, appSchema, &update)
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			_, err = w.Apps.Update(ctx, update.UpdateAppRequest)
+			if err != nil {
+				return err
+			}
+			if d.HasChanges("source_code_path", "mode") {
```

Comment:
See my comment above, I think we should remove app deployment altogether for now